src/coreclr/jit/instr.cpp

```diff
@@ -1096,37 +1113,110 @@ void CodeGen::inst_RV_TT_IV(instruction ins, emitAttr attr, regNumber reg1, GenT
+            {
+                IsEmbBroadcastEnabled = true;
+            }
+            break;
+        }
+
+        default:
+            break;
+    }
+
+    return IsEmbBroadcastCompatible && IsEmbBroadcastEnabled;
+}
+#endif //  TARGET_XARCH && FEATURE_HW_INTRINSICS
+
 //------------------------------------------------------------------------
 // inst_RV_RV_TT: Generates an instruction that takes 2 operands:
 //                a register operand and an operand that may be in memory or register
 //                the result is returned in register
 //
 // Arguments:
-//    ins       -- The instruction being emitted
-//    size      -- The emit size attribute
-//    targetReg -- The target register
-//    op1Reg    -- The first operand register
-//    op2       -- The second operand, which may be a memory node or a node producing a register
-//    isRMW     -- true if the instruction is RMW; otherwise, false
-//
-void CodeGen::inst_RV_RV_TT(
-    instruction ins, emitAttr size, regNumber targetReg, regNumber op1Reg, GenTree* op2, bool isRMW)
+//    ins          -- The instruction being emitted
+//    size         -- The emit size attribute
+//    targetReg    -- The target register
+//    op1Reg       -- The first operand register
+//    op2          -- The second operand, which may be a memory node or a node producing a register
+//    isRMW        -- true if the instruction is RMW; otherwise, false
+//    simdBaseType -- the base data type for this intrinsic.
+void CodeGen::inst_RV_RV_TT(instruction ins,
+                            emitAttr    size,
+                            regNumber   targetReg,
+                            regNumber   op1Reg,
+                            GenTree*    op2,
+                            bool        isRMW,
+                            var_types   simdBaseType)
 {
     emitter* emit = GetEmitter();
     noway_assert(emit->emitVerifyEncodable(ins, EA_SIZE(size), targetReg));
 
     // TODO-XArch-CQ: Commutative operations can have op1 be contained
     // TODO-XArch-CQ: Non-VEX encoded instructions can have both ops contained
 
-    OperandDesc op2Desc = genOperandDesc(op2);
+    OperandDesc op2Desc     = genOperandDesc(op2);
+    insOpts     instOptions = INS_OPTS_NONE;
+#if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
+    bool IsEmbBroadcast = CodeGenInterface::IsEmbeddedBroadcastEnabled(ins, op2);
+    if (IsEmbBroadcast)
+    {
+        instOptions = INS_OPTS_EVEX_b;
+        if (op2->OperIs(GT_CNS_VEC))
+        {
+            switch (simdBaseType)
+            {
+                case TYP_FLOAT:
+                {
+                    float scalar = static_cast<float>(op2->AsVecCon()->gtSimd32Val.f32[0]);
+                    op2Desc      = OperandDesc(emit->emitFltOrDblConst(*reinterpret_cast<float*>(&scalar), EA_4BYTE));
+                    break;
+                }
+
+                default:
+                    break;
+            }
+        }
+    }
```

Comment:
This is going to double emit the constant. Once in `genOperandDesc` and once here.

We're going to need to move this logic into `genOperandDesc`, possibly by passing in enough data so that it can check, or we'll have to ensure that `Lowering` recognizes this case and replaces the `GT_CNS_VEC` with `BroadcastScalarToVectorXXX(CreateScalarUnsafe(DBL_CNS))`